### ISSUE TITLE
Please consider exposing maxBondMatchPairs param in rdRascalMCES.RascalOptions()

### ISSUE THREAD

**Issue by nate-russell** [rdkit/rdkit#7054]
rdRascalMCES.FindMCES is great but the maxBondMatchPairs = 1000; limit is making it hard to work with lipids.  A warning about exceeding 1000 might also be warranted. 

https://github.com/rdkit/rdkit/blob/73b4da2a44a297758def89dd4f9eda8cf0b4e907/Code/GraphMol/RascalMCES/RascalOptions.h  has an option for maxBondMatchPairs but this option is not exposed in the wrapper https://github.com/rdkit/rdkit/blob/73b4da2a44a297758def89dd4f9eda8cf0b4e907/Code/GraphMol/RascalMCES/Wrap/rdRascalMCES.cpp

The option to modify the limit puts more control in the user's hands. 

```
import rdkit
print('rdkit version: ',rdkit.__version__)
import sys
print(sys.version)

from rdkit import Chem
from rdkit.Chem import rdRascalMCES

def rascal_wrapper(mola,molb):
    opts = rdRascalMCES.RascalOptions()
    opts.similarityThreshold = 0.0
    opts.returnEmptyMCES = True
    opts.singleLargestFrag = True
    opts.allBestMCESs = True
    opts.maxBondMatchPairs = 10000
    opts.completeAromaticRings = False
    opts.timeout = -1
    results = rdRascalMCES.FindMCES(mola, molb,opts)
    print(f'Results {results}')
    print(f"N Results: {len(results)}")

    print("tier1sim:",results[0].tier1Sim,"tier2sim:",results[0].tier2Sim)
    
    for i,res in enumerate(results):
        print(f"Res {i}")
        print(f'MCES SMARTS : {res.smartsString}')
        print(f'Matching Bonds : {res.bondMatches()}')
        print(f'Matching Atoms : {res.atomMatches()}')


# These are the Lipids i actually wanted to compare
print('\nLipids I care about')
real_mol_1 =  Chem.MolFromSmiles('CCCCCCCC\C=C/CCCCCCCCNC(=O)C(CCCCCOC(=O)CCCCCCC\C=C/CCCCCCCC)NCCCN(C)CCCN')
real_mol_2 =  Chem.MolFromSmiles('CCCCCCCCCCC(=O)OCCCCCC(NCCCN(C)CCCN)C(=O)NCCCCCCCC\C=C/CCCCCCCC')
rascal_wrapper(real_mol_1,real_mol_2)


# Larger heteroatom ring systems seem to work ok so I think my issue is related to carbon chains
# This was an example molecule pair I made to test the limits
print('\nReturns results')
gets_results_1 =  Chem.MolFromSmiles('CCCC=CCCCC=CCCCCCCCCCCCCCCCCCCCC1CNCCC1')
gets_results_2 =  Chem.MolFromSmiles('CCCC=CCCCCC=CCCCCCCCCCCCCCCCCCCC1CNCCC1')
rascal_wrapper(gets_results_1,gets_results_2)

# I add one extra carbon and I hit the limit
print('\nNo Results: 1 carbon too long resulting in Too many matching bond pairs (1032) so can\'t continue.')
too_long_1 =  Chem.MolFromSmiles('CCCC=CCCCC=CCCCCCCCCCCCCCCCCCCCCC1CNCCC1')
too_long_2 =  Chem.MolFromSmiles('CCCC=CCCCCC=CCCCCCCCCCCCCCCCCCCCC1CNCCC1')
rascal_wrapper(too_long_1,too_long_2)`
```

> rdkit version:  2023.09.3
> 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:27:34) [MSC v.1937 64 bit (AMD64)]
> 
> Lipids I care about
> Too many matching bond pairs (1586) so can't continue.
> Results [<rdkit.Chem.rdRascalMCES.RascalResult object at 0x000002610BE09900>]
> N Results: 1
> tier1sim: 0.8761061946902655 tier2sim: 0.8761061946902655
> Res 0
> MCES SMARTS :
> Matching Bonds : []
> Matching Atoms : []
> 
> Returns results
> Results [<rdkit.Chem.rdRascalMCES.RascalResult object at 0x000002610BE09900>]
> N Results: 1
> tier1sim: 1.0 tier2sim: 1.0
> Res 0
> MCES SMARTS : CCCCCCCCCCCCCCCCCCCC1CNCCC1
> Matching Bonds : [(10, 10), (11, 11), (12, 12), (13, 13), (14, 14), (15, 15), (16, 16), (17, 17), (18, 18), (19, 19), (20, 20), (21, 21), (22, 22), (23, 23), (24, 24), (25, 25), (26, 26), (27, 27), (28, 28), (29, 29), (30, 30), (31, 31), (32, 32), (33, 33), (34, 34)]
> Matching Atoms : [(10, 10), (11, 11), (12, 12), (13, 13), (14, 14), (15, 15), (16, 16), (17, 17), (18, 18), (19, 19), (20, 20), (21, 21), (22, 22), (23, 23), (24, 24), (25, 25), (26, 26), (27, 27), (28, 28), (29, 29), (30, 30), (31, 31), (32, 32), (33, 33), (34, 34)]
> 
> No Results: 1 carbon too long resulting in Too many matching bond pairs (1032) so can't continue.
> Too many matching bond pairs (1032) so can't continue.
> Results [<rdkit.Chem.rdRascalMCES.RascalResult object at 0x000002610BE09900>]
> N Results: 1
> tier1sim: 1.0 tier2sim: 1.0
> Res 0
> MCES SMARTS :
> Matching Bonds : []
> Matching Atoms : []
>

**Comment by DavidACosgrove** [comment]
Hi Nate.  That was an oversight.  Apologies.   I have just made a pull request exposing the option as requested.  You should use it with caution, as memory use goes with the square of the number of matching bond pairs.  Your value of 10000 above will probably crash your machine if it actually gets up to that sort of number of bond pair matches.

**Comment by greglandrum** [comment]
Fixed in #7073
